app/controllers/stages_controller.rb

```diff
@@ -102,4 +99,12 @@ def stage_params
   def find_stage
     @stage = current_project.stages.find_by_param!(params[:id])
   end
+
+  def template_param
+    @template_param ||= ActiveRecord::Type::Boolean.new.type_cast_from_database(stage_params.delete("template"))
+  end
+
+  def update_template
+    template_param ? @stage.project.template!(@stage) : @stage.project.template!
```

Comment:
can you make that more intuitive ? maybe keyword args ?

On Tue, Aug 16, 2016 at 1:59 PM, Adam Panzer <notifications@github.com>
wrote:

> In app/controllers/stages_controller.rb
> <https://github.com/zendesk/samson/pull/1198#discussion_r75018949>:
>
> > @@ -102,4 +99,12 @@ def stage_params
> >    def find_stage
> >      @stage = current_project.stages.find_by_param!(params[:id])
> >    end
> > +
> > +  def template_param
> > +    @template_param ||= ActiveRecord::Type::Boolean.new.type_cast_from_database(stage_params.delete("template"))
> > +  end
> > +
> > +  def update_template
> > +    template_param ? @stage.project.template!(@stage) : @stage.project.template!
>
> eh?
>
> template!(nil) => Just unset the template stage for the project.
> template!() => Unset the existing template for the project (if there is
> one) and set the one given.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1198/files/d43bdda94cc05d08fa093f7c694b09953335fe89#r75018949>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ6zi1cgXp8DRM4SvuCDFOyMErZL8ks5qgiSmgaJpZM4Jhtog>
> .
>